specification/subscription/resource-manager/Microsoft.Subscription/preview/2019-10-01-preview/subscriptions.json

```diff
@@ -392,6 +392,175 @@
           }
         ]
       }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Subscription/targetDirectory/default": {
+      "put": {
+        "description": "Change target directory.",
+        "operationId": "Subscription_ChangeTargetDirectory",
+        "x-ms-examples": {
+          "ChangeTargetDirectory": {
+            "$ref": "./examples/ChangeTargetDirectory.json"
+          }
+        },
+        "responses": {
+          "200": {
+            "description": "Normal response for a successful query.",
+            "schema": {
+              "$ref": "#/definitions/ChangeDirectoryResponse"
```

Comment:
request and respnse can be the same model. If anything is not settable in request but service generate, it can be marked as read only.

Reply:
Made request and response same